lib/mongo/server/connection_pool.rb

```diff
@@ -729,6 +747,16 @@ def self.finalize(available_connections, pending_connections, populator)
 
       private
 
+      def next_available_connection(service_id: nil)
+        if service_id
+          @available_connections.detect do |conn|
```

Comment:
This does not remove the connection from available connections, I'll need to repair that.